device.c

```diff
@@ -62,25 +63,49 @@ static void device_lock(struct device *device)
 	if (fd >= 0)
 		close(fd);
 
-	fd = open(lock, O_RDONLY | O_CLOEXEC);
+	fd = open(lock, O_RDWR | O_CLOEXEC);
 	if (fd < 0)
 		err(1, "failed to open lockfile %s", lock);
 
+	/* Read current user out of the lockfile if there is one */
+	n = read(fd, user, sizeof(user)-1);
+	if (n < 0)
+		err(1, "failed to read lockfile %s", lock);
+	/* Strip newline */
+	if (n)
+		user[n-1] = '\0';
+
 	while (1) {
 		char c;
 
 		n = flock(fd, LOCK_EX | LOCK_NB);
 		if (!n)
-			return;
+			break;
 
-		warnx("board is in use, waiting...");
+		warnx("board is in use by %s, waiting...", user);
```

Comment:
User is not a constant, it needs to be re-read constantly when printing the message. Consider `userB` "stealing" the board while you are waiting for it to be freed by `userA`. The message will still print `userA`, while you should be pinging `userB`.